diff -Naurp GPTL/gptl.c patchvGPTL/gptl.c
--- GPTL/gptl.c	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/gptl.c	2014-07-30 05:49:35.894605380 -0700
@@ -396,7 +396,10 @@ int GPTLinitialize (void)
   int t;          /* thread index */
   double t1, t2;  /* returned from underlying timer */
   static const char *thisfunc = "GPTLinitialize";
-
+#ifdef HAVE_LIBXML2
+  //Initialize xml parser function
+  GPTLxmlparser();
+#endif
   if (initialized)
     return GPTLerror ("%s: has already been called\n", thisfunc);
 
diff -Naurp GPTL/macros.make.bluegene patchvGPTL/macros.make.bluegene
--- GPTL/macros.make.bluegene	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.bluegene	2014-07-30 05:49:35.894605380 -0700
@@ -165,3 +165,12 @@ ifeq ($(HAVE_MPI),yes)
     HAVE_IARGCGETARG = no
   endif
 endif
+
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
diff -Naurp GPTL/macros.make.lahey patchvGPTL/macros.make.lahey
--- GPTL/macros.make.lahey	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.lahey	2014-07-30 05:49:35.894605380 -0700
@@ -152,6 +152,14 @@ ifeq ($(HAVE_MPI),yes)
   endif
 endif
 
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
 # Whether system functions backtrace() and backtrace_symbols() exist. Setting HAVE_BACKTRACE=yes
 # will enable auto-generation of function name when auto-profiling and GPTLdopr_memusage has been
 # enabled at run-time. If HAVE_BACKTRACE=no, function address will be printed instead.
diff -Naurp GPTL/macros.make.linux patchvGPTL/macros.make.linux
--- GPTL/macros.make.linux	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.linux	2014-07-30 05:49:35.894605380 -0700
@@ -152,6 +152,14 @@ ifeq ($(HAVE_MPI),yes)
   endif
 endif
 
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
 # Whether system functions backtrace() and backtrace_symbols() exist. Setting HAVE_BACKTRACE=yes
 # will enable auto-generation of function name when auto-profiling and GPTLdopr_memusage has been
 # enabled at run-time. If HAVE_BACKTRACE=no, function address will be printed instead.
diff -Naurp GPTL/macros.make.macos patchvGPTL/macros.make.macos
--- GPTL/macros.make.macos	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.macos	2014-07-30 05:49:35.894605380 -0700
@@ -153,6 +153,14 @@ ifeq ($(HAVE_MPI),yes)
   endif
 endif
 
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
 # Whether system functions backtrace() and backtrace_symbols() exist. Setting HAVE_BACKTRACE=yes
 # will enable auto-generation of function name when auto-profiling and GPTLdopr_memusage has been
 # enabled at run-time. If HAVE_BACKTRACE=no, function address will be printed instead.
diff -Naurp GPTL/macros.make.pgi patchvGPTL/macros.make.pgi
--- GPTL/macros.make.pgi	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.pgi	2014-07-30 05:49:35.894605380 -0700
@@ -152,6 +152,14 @@ ifeq ($(HAVE_MPI),yes)
   endif
 endif
 
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
 # Whether system functions backtrace() and backtrace_symbols() exist. Setting HAVE_BACKTRACE=yes
 # will enable auto-generation of function name when auto-profiling and GPTLdopr_memusage has been
 # enabled at run-time. If HAVE_BACKTRACE=no, function address will be printed instead.
diff -Naurp GPTL/macros.make.xeon patchvGPTL/macros.make.xeon
--- GPTL/macros.make.xeon	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.xeon	2014-07-30 05:49:35.894605380 -0700
@@ -152,6 +152,14 @@ ifeq ($(HAVE_MPI),yes)
   endif
 endif
 
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
 # Whether system functions backtrace() and backtrace_symbols() exist. Setting HAVE_BACKTRACE=yes
 # will enable auto-generation of function name when auto-profiling and GPTLdopr_memusage has been
 # enabled at run-time. If HAVE_BACKTRACE=no, function address will be printed instead.
diff -Naurp GPTL/macros.make.xeonphi patchvGPTL/macros.make.xeonphi
--- GPTL/macros.make.xeonphi	2014-07-30 05:16:12.362605619 -0700
+++ patchvGPTL/macros.make.xeonphi	2014-07-30 05:49:35.894605380 -0700
@@ -152,6 +152,14 @@ ifeq ($(HAVE_MPI),yes)
   endif
 endif
 
+#If LibXML2 is present, we can specify instrumentation details in xml file dynamically without 
+#recompiling source
+HAVE_LIBXML2 = yes
+  ifeq ($(HAVE_LIBXML2),yes)
+    LIBXML2_INCFLAGS = -I/usr/include/libxml2
+    LIBXML2_LIBFLAGS = -L/usr/local/lib64 -lxml2
+  endif
+
 # Whether system functions backtrace() and backtrace_symbols() exist. Setting HAVE_BACKTRACE=yes
 # will enable auto-generation of function name when auto-profiling and GPTLdopr_memusage has been
 # enabled at run-time. If HAVE_BACKTRACE=no, function address will be printed instead.
diff -Naurp GPTL/Makefile patchvGPTL/Makefile
--- GPTL/Makefile	2014-07-30 05:16:12.358605619 -0700
+++ patchvGPTL/Makefile	2014-07-30 05:49:35.894605380 -0700
@@ -7,8 +7,8 @@ else
 endif
 
 null =
-OBJS = gptl.o util.o get_memusage.o print_memusage.o gptl_papi.o pmpi.o getoverhead.o \
-       hashstats.o memstats.o pr_summary.o print_rusage.o
+OBJS = gptl.o util.o get_memusage.o print_memusage.o gptl_papi.o pmpi.o setxmlConfig.o \
+       getoverhead.o hashstats.o memstats.o pr_summary.o print_rusage.o
 
 ifeq ($(ENABLE_PMPI),yes)
   CFLAGS += -DENABLE_PMPI -DMPI_STATUS_SIZE_IN_INTS=$(MPI_STATUS_SIZE_IN_INTS)
@@ -162,6 +162,7 @@ gptl_papi.o: gptl.h private.h
 process_namelist.o: process_namelist.F90 gptl.inc
 gptlf.o: gptlf.F90
 pmpi.o: gptl.h private.h
+setxmlConfig.o: gptl.h private.h
 getoverhead.o: private.h
 hashstats.o: private.h
 memstats.o: private.h
diff -Naurp GPTL/Makefile.in patchvGPTL/Makefile.in
--- GPTL/Makefile.in	2014-07-30 05:16:12.358605619 -0700
+++ patchvGPTL/Makefile.in	2014-07-30 05:49:35.894605380 -0700
@@ -6,7 +6,7 @@ LIBDIR   = @prefix@/lib
 INCDIR   = @prefix@/include
 MANDIR   = @prefix@/man/man3
 OBJS     = f_wrappers.o gptl.o util.o threadutil.o get_memusage.o \
-           print_memusage.o gptl_papi.o
+           print_memusage.o gptl_papi.o setxmlConfig.o
 AR       = @AR@
 
 @SET_MAKE@
@@ -41,3 +41,4 @@ gptl.o: gptl.c gptl.h private.h
 util.o: util.c gptl.h private.h
 threadutil.o: threadutil.c gptl.h private.h
 gptl_papi.o: gptl_papi.c gptl.h private.h
+setxmlConfig.o: gptl.h private.h
diff -Naurp GPTL/suggestions patchvGPTL/suggestions
--- GPTL/suggestions	2014-07-30 05:16:12.366605619 -0700
+++ patchvGPTL/suggestions	2014-07-30 05:49:35.894605380 -0700
@@ -4133,7 +4133,7 @@ if ${ac_cv_lib_pthread_pthread_mutex_ini
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpthread  $LIBS"
+LIBS="-lpthread -lxml2  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
